From 510e1f0e67033cb08a874f60800e390fd516b3f7 Mon Sep 17 00:00:00 2001
From: Gwenael Treuveur <gwenael.treuveur@foss.st.com>
Date: Thu, 14 Nov 2024 15:57:42 +0100
Subject: [PATCH 1382/1695] dts: stm32: add OEM rproc encryption key for
 stm32mp23

Add new OTP entry for remoteproc encryption key.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 3f5023cce9a821f25fff73bed864caf13734a37e dts: stm32: add OEM rproc encryption key for stm32mp23]
Change-Id: I28409ce1c527cdc339a7d89606ae74555cfb699d
Signed-off-by: Gwenael Treuveur <gwenael.treuveur@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/418655
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Tested-by: Gwenael TREUVEUR <gwenael.treuveur@st.com>
Reviewed-by: Gwenael TREUVEUR <gwenael.treuveur@st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
---
 core/arch/arm/dts/stm32mp231.dtsi | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp231.dtsi b/core/arch/arm/dts/stm32mp231.dtsi
index 0d521d3be..78dee4da3 100644
--- a/core/arch/arm/dts/stm32mp231.dtsi
+++ b/core/arch/arm/dts/stm32mp231.dtsi
@@ -2780,6 +2780,12 @@
 				st,non-secure-otp-provisioning;
 			};
 
+			oem_rproc_enc_key: oem_rproc_enc_key@540 {
+				/* 0x540 -> otp336 */
+				reg = <0x540 0x20>;
+				st,non-secure-otp-provisioning;
+			};
+
 			oem_enc_key_2: otp360@5a0 {
 				reg = <0x5a0 0x10>;
 				st,non-secure-otp-provisioning;
-- 
2.47.2

